core/src/main/scala/com/microsoft/azure/synapse/ml/train/ComputeModelStatistics.scala

```diff
@@ -281,7 +281,7 @@ class ComputeModelStatistics(override val uid: String) extends Transformer
     // Calculate confusion matrix and output it as DataFrame
     // TODO: We call cache in order to avoid a bug with catalyst where CMS seems to get stuck in a loop
     // For future spark upgrade past 2.2.0, we should try to see if the cache() call can be removed
-    dataset.select(col(scoredLabelsColumnName), col(labelColumnName))
+    dataset.select(col(scoredLabelsColumnName).cast(DoubleType), col(labelColumnName))
       .cache()
```

Comment:
might be interesting to see if this cache call can indeed finally be removed, I remember this was a pretty nasty bug

Reply:
Since I don't know anything about this bug, I'm not sure how to validate it. Do you have any ideas what this bug is?